src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_base_decorator.py

```diff
@@ -24,9 +24,9 @@ def test_validate_decorator_mode(self):
         self.assertEqual(validate_decorator_mode(DecoratorMode.CREATE), True)
         self.assertEqual(validate_decorator_mode(DecoratorMode.UPDATE), True)
         self.assertEqual(validate_decorator_mode(DecoratorMode), False)
-        self.assertEqual(validate_decorator_mode(1), False)
+        # self.assertEqual(validate_decorator_mode(1), False)
```

Comment:
In 3.12, the emum's value is also `in` Enum.
This is the result in 3.11:
```
>>> from enum import Enum
>>> class DecoratorMode(Enum):
...     """Enumerations used to distinguish whether to handle creation or update.
...     """
...     CREATE = 1
...     UPDATE = 2
...
>>> 1 in DecoratorMode
<stdin>:1: DeprecationWarning: in 3.12 __contains__ will no longer raise TypeError, but will return True or
False depending on whether the value is a member or the value of a member
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\enum.py", line 742, in __contains__
    raise TypeError(
TypeError: unsupported operand type(s) for 'in': 'int' and 'EnumType'

>>> True in DecoratorMode
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\enum.py", line 742, in __contains__
    raise TypeError(
TypeError: unsupported operand type(s) for 'in': 'bool' and 'EnumType'

>>> "1" in DecoratorMode
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\enum.py", line 742, in __contains__
    raise TypeError(
TypeError: unsupported operand type(s) for 'in': 'str' and 'EnumType'
```
In 3.12
```
>>> 1 in DecoratorMode
True
>>> True in DecoratorMode
True
>>> "1" in  DecoratorMode
False
```

>Changed in version 3.12: Before Python 3.12, a `TypeError` is raised if a non-Enum-member is used in a containment check. --https://docs.python.org/3/library/enum.html#enum.EnumType.__contains__

Reply:
Fixing in #29517